modules/programs/zsh.nix

```diff
@@ -37,6 +37,26 @@ let
     };
   };
 
+  pluginModule = types.submodule ({ config, ... }: {
+    options = {
+      src = mkOption {
+        type = types.path;
+        description = "Path to the plugin folder. Will be added to <envar>fpath</envar> and <envar>PATH</envar>.";
+      };
+
+      name = mkOption {
+        type = types.str;
+        description = "The name of the plugin. Use <option>file</option> instead if the script name does not follow convention.";
+      };
+
+      file = mkOption {
+        type = types.str;
+        default = "${config.name}.plugin.zsh";
```

Comment:
Like @uvNikita mentioned this will cause problems because Nix will attempt to expand this during evaluation even if the modules is not used in the configuration, e.g., for generating the `home-configuration.nix` man page.

Instead I think you can add a `config` section:

```
pluginModule = types.submodule ({ config, ... }: {
  options = {
    …
  };

  config = {
    file = mkDefault name;
  };
};
```

Reply:
Amended the fix